### PR TITLE
PUBDEV-7020: Updated Target Encoding data munging topic

### PR DESCRIPTION
**Pull request by angela0xdata**
Updated TE topic based on current api
Driveby fix: Fixed formatting of "transform" parameter list in the Python module docs.

### CODE REVIEW

**Comment by deil87** [comment]
@ledell @angela0xdata Titanic dataset is able to show benefits of using TE but to what extent depends on TE's parameters, seeds and type of algo (e.g. GBM). I think that once TE integration is done we will be able to find those optimal parameters. But for now I believe it is not that difficult to find decent parameters by playing a bit with them. I can help with this if it's needed, let me know.

**Comment by meganjkurka** [comment]
Can't the same example I had be used with the new API rather than finding a new dataset?

**Comment by angela0xdata** [comment]
@meganjkurka @ledell I ran into (cough *user*) errors when I tried replacing the previous housing example. I can try again, but I'm not sure it will make it into a release tomorrow.

File: h2o-docs/src/product/data-munging/target-encoding.rst [hunk 1]
```diff
@@ -1,461 +1,160 @@
...
+    fold_column = "pclass"
+    data_leakage_handling = "k_fold"
 
-   .. code-tab:: python
+    # Train a TE model
+    titanic_te = H2OTargetEncoderEstimator(fold_column=fold_column,
+                                           data_leakage_handling=data_leakage_handling)
+    te_model = titanic_te.train(x=encoded_columns, 
+                                y=response,
+                                training_frame=titanic)
```
Comment (deil87) [inline h2o-docs/src/product/data-munging/target-encoding.rst:152]: @angela0xdata @ledell  we should not use whole `titanic` dataset for training. I'm working right now on the extended demo ( with comparison) . Not sure if I will make it in time, but I will try.

Comment (deil87) [reply h2o-docs/src/product/data-munging/target-encoding.rst:152]: @angela0xdata @ledell found optimal values, prepared tests ( for both an old api and a new api as I needed to compute aucs with the old api due to a bug in the new one). Transferring code to .rst file.

**Comment by angela0xdata** [comment]
@deil87 I'm getting the following error when I try to build the GBM with the the new te columns. (I'm just copying and pasting):

```
>>> gbm_with_te.train(x=x_with_te, y=response, training_frame=train_te)
gbm Model Build progress: |▋ (failed)                                                         |   1%
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/site-packages/h2o/estimators/estimator_base.py", line 115, in train
    self._train(parms, verbose=verbose)
  File "/usr/local/lib/python3.7/site-packages/h2o/estimators/estimator_base.py", line 202, in _train
    job.poll(poll_updates=self._print_model_scoring_history if verbose else None)
  File "/usr/local/lib/python3.7/site-packages/h2o/job.py", line 78, in poll
    "\n{}".format(self.job_key, self.exception, self.job["stacktrace"]))
OSError: Job with key $03017f00000132d4ffffffff$_ab025bda3984f29aae82b1b6008d4442 failed with an exception: java.lang.IllegalArgumentException: class water.fvec.Frame _a5664ff30b75c5f2ea4812db4470455c is being created;  Unable to read it now.
stacktrace: 
java.lang.IllegalArgumentException: class water.fvec.Frame _a5664ff30b75c5f2ea4812db4470455c is being created;  Unable to read it now.
	at water.Lockable$ReadLock.atomic(Lockable.java:151)
	at water.Lockable$ReadLock.atomic(Lockable.java:145)
	at water.TAtomic.atomic(TAtomic.java:17)
	at water.Atomic.compute2(Atomic.java:56)
	at water.Atomic.fork(Atomic.java:39)
	at water.Atomic.invoke(Atomic.java:31)
	at water.Lockable.read_lock(Lockable.java:140)
	at hex.Model$Parameters.read_lock_frames(Model.java:362)
	at hex.ModelBuilder$Driver.compute2(ModelBuilder.java:245)
	at water.H2O$H2OCountedCompleter.compute(H2O.java:1551)
	at jsr166y.CountedCompleter.exec(CountedCompleter.java:468)
	at jsr166y.ForkJoinTask.doExec(ForkJoinTask.java:263)
	at jsr166y.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:974)
	at jsr166y.ForkJoinPool.runWorker(ForkJoinPool.java:1477)
	at jsr166y.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:104)
```
This happens in Python 2.7 and 3.7. 
I verified that the train_te dataset does include those columns:
```
>>> train_te
  pclass    survived  name                                               sex         age    sibsp    parch    ticket      fare  cabin    embarked      boat    body  home.dest                          kfold_column    cabin_te    embarked_te    home.dest_te
```
I don't get this error when running the baseline example with "train" training frame.

**Comment by deil87** [comment]
@angela0xdata I'm not a part of this company anymore. Please reach out to someone else. But it looks like the error that was fixed by @Pscheidl recently. Maybe you need to update/rebuild dependencies.

**Comment by angela0xdata** [comment]
@deil87 whoops, sorry! 
@Pscheidl was this fix merged into master? (I updated this branch from master earlier today, so it should have all recent fixes.)

**Comment by angela0xdata** [comment]
Merging so that the fix from https://0xdata.atlassian.net/browse/PUBDEV-7434 gets included.

**Pull request merged**